planner/core/planbuilder.go

```diff
@@ -1518,6 +1518,47 @@ func (b *PlanBuilder) buildDDL(node ast.DDLNode) (Plan, error) {
 			table:     v.Table.Name.L,
 			err:       nil,
 		})
+		for _, spec := range v.Specs {
+			if spec.Tp == ast.AlterTableRenameTable {
+				b.visitInfo = append(b.visitInfo, visitInfo{
+					privilege: mysql.DropPriv,
+					db:        v.Table.Schema.L,
+					table:     v.Table.Name.L,
+					err:       nil,
+				})
+				b.visitInfo = append(b.visitInfo, visitInfo{
+					privilege: mysql.CreatePriv,
+					db:        spec.NewTable.Schema.L,
+					table:     spec.NewTable.Name.L,
+					err:       nil,
+				})
+				b.visitInfo = append(b.visitInfo, visitInfo{
```

Comment:
you can make `plan.appendVisitInfo` public, and use this function to make code clearer.

Reply:
Done.
And `appendVisitInfo` is in `planner/core`, we do not need to make it exported.